cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -282,7 +283,11 @@ def get_vr_address():
         return latest_address
 
     # Try dhcp lease files next...
-    lease_file = dhcp.IscDhclient.get_latest_lease()
+    distro = stages.Init(reporter=True).distro  # detect OS distro
```

Comment:
Driveby comment:  Why call Init()? `__init__()` already got an instance of distro, lets just save a reference to it in an instance variable and use that rather than calling `Init()`.

Reply:
Indeed, I noticed that the Distro object is already passed into `sources.DataSource.__init__()` as an argument. Thus, no need to obtain the Distro object once more.
Switched to using that. 

Does the rest look OK so I can implement it for the other distros?